middle_end/flambda2/types/env/typing_env.ml

```diff
@@ -1309,11 +1311,13 @@ end = struct
       | Naked_int16 _ | Naked_int32 _ | Naked_int64 _ | Naked_vec128 _
       | Naked_vec256 _ | Naked_vec512 _ | Naked_nativeint _ | Rec_info _
       | Region _ ->
-        assert false
+        if expect_value
+        then Misc.fatal_errorf "Got type %a, but expected value" TG.print ty
+        else Value_unknown
```

Comment:
I'd like to get @mshinwell to take a look at this diff. In particular, it would be good if he could confirm `Value_unknown` is OK for all unboxed types and doesn't actually imply `value` in the layout sense.  If it does, I think we may need to extend `value_approximation` with new approximations?

Reply:
This is for @lthls or @bclement-ocp

Reply:
`Value_unknown` implies `Value` layout, so it is wrong for unboxed types. But I'm a bit surprised we didn't hit this case earlier: any symbol with a mixed shape, imported in a compilation unit compiled in Classic mode, should have triggered the assertion.
For this PR, I would suggest detecting whether any of the fields are not of kind value and returning `Value_unknown` for the whole block in that case (just like the `Float_record` case), at least until we find a better solution.